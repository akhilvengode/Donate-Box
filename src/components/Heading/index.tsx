import classnames from "classnames";
import styles from "./index.module.css";

type PropType = {
  className?: string;
  textAlign?: "left" | "center" | "right";
  varient?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
};

const Heading: React.FC<PropType> = ({
  children,
  className,
  textAlign = "left",
  varient = "h1",
}) => {
  const Heading = varient;
  return (
    <Heading
      className={classnames(
        styles.heading,
        styles[`heading--${textAlign}`],
        className
      )}
    >
      {children}
    </Heading>
  );
};

export default Heading;
