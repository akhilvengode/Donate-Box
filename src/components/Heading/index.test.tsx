import { shallow } from "enzyme";
import Heading from ".";

describe("Testing Heading Component", () => {
  const wrapper = shallow(<Heading>Test Head</Heading>);

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render h1 element as default", () => {
    expect(wrapper.find("h1")).toHaveLength(1);
  });

  it("should render the right varient", () => {
    const wrapper = shallow(<Heading varient="h3">Testing h3</Heading>);
    expect(wrapper.find("h3")).toHaveLength(1);
  });
});
