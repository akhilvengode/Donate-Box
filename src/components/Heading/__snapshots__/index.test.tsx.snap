// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Heading Component should match the snapshot 1`] = `
<h1
  className="heading heading--left"
>
  Test Head
</h1>
`;
